### DISCUSSION TITLE
How to find reason for occasional long query

### DISCUSSION THREAD

**Discussion by jshen28** [sqlalchemy/sqlalchemy#8764] [Usage Questions]
Hello,

  I am trying to find the root cause for a occasional long query. The behavior is `query.all()` occasionally returns longer than 3 seconds. I add a hook before and after execute, the query execution time is less than 0.01 second. The data is around 100 rows joining 6 tables. 
Because it is not constantly reproducible, I would like to ask if there is a way to identity the root cause quickly. Thank you.

Best,
Norman

**Comment by CaselIT** [comment]
Hi,

Are you asking how to find a query that takes a long time to run or you already know which one takes a long time to run and want help debugging?

**Comment by jshen28** [comment]
Hi,

   Thank you for the reply. I'd like to ask beyond execution time in DB
what else could causes fluctuation of `query.all`. I am trying to debug a
weird situation where sometimes _close_impl takes an extra 3 seconds to
complete. Thank you.

Best,
Norman

On Mon, Nov 7, 2022 at 3:50 PM Federico Caselli ***@***.***>
wrote:

> Hi,
>
> Are you asking how to find a query that takes a long time to run or you
> already know which one takes a long time to run and want help debugging?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sqlalchemy/sqlalchemy/discussions/8764#discussioncomment-4073848>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADH3XA4IWZU2FQMQQGEMYLLWHCYETANCNFSM6AAAAAARYX3R3E>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Best Regards,

Jiatong Shen
